src/main/java/org/maxgamer/quickshop/shop/ContainerShop.java

```diff
@@ -387,6 +413,7 @@ public void delete() {
     @Override
     public void delete(boolean memoryOnly) {
         Util.ensureThread(false);
+        ContainerShop neighbor = getAttachedShop();
```

Comment:
Emmmm, I think we should put the variable to where we actually need to use as nearly as possible.

Reply:
The reason for it being up there is so that it save it from deletion. If I were to put it in the if statement at the end, it wouldn't work since the shop would have been deleted at that point.

Reply:
So please add a comment for it

Reply:
Added!